src/hal/src/command/renderpass.rs

```diff
@@ -17,39 +18,9 @@ pub enum SubpassContents {
     SecondaryBuffers,
 }
 
-///
-pub struct RenderPassInlineEncoder<'a, B: Backend>(pub(crate) &'a mut B::CommandBuffer)
-where B::CommandBuffer: 'a;
-
-impl<'a, B: Backend> RenderPassInlineEncoder<'a, B> {
-    ///
-    pub fn new<C, T>(
-        cmd_buffer: &'a mut CommandBuffer<B, C>,
-        render_pass: &B::RenderPass,
-        frame_buffer: &B::Framebuffer,
-        render_area: Rect,
-        clear_values: T,
-    ) -> Self
-    where
-        C: Supports<Graphics>,
-        T: IntoIterator,
-        T::Item: Borrow<ClearValue>,
-    {
-        cmd_buffer.raw.begin_renderpass(
-            render_pass,
-            frame_buffer,
-            render_area,
-            clear_values,
-            SubpassContents::Inline);
-        RenderPassInlineEncoder(cmd_buffer.raw)
-    }
-
-    ///
-    pub fn next_subpass_inline(self) -> Self {
-        self.0.next_subpass(SubpassContents::Inline);
-        self
-    }
-
+// This macro contains all methods for all commands submittable during a subpass.
+// It is used to implement the identical portions of RenderPassInlineEncoder and SubpassCommandBuffer's API.
+macro_rules! add_subpass_commands { ($buf:ident) => {
```

Comment:
can we share the implementations without meta-programming (please?), maybe like this:
```rust
struct RenderSubpassCommon {
  // common methods
}
impl Deref<RenderSubpassCommon> for RenderPassInlineEncoder {..}
impl Deref<RenderSubpassCommon> for SubpassCommandBuffer {..}
```

Reply:
Would this work without requiring `use RenderSubpassCommon`?

Reply:
Sure, since `RenderSubpassCommon` is a struct, no a trait

Reply:
Hm, not sure how to implement this when `RenderPassInlineEncoder` borrows its command buffer but `SubpassCommandBuffer` owns its.